File: Models/UserProfile.cs
namespace DevoteesAnusanga.Models
{
    public class UserProfile
    {
        public Guid Id { get; set; }
        public Guid UserId { get; set; }

        public string Name { get; set; }
        public string Email { get; set; }
        public string Phone { get; set; }

        public string Country { get; set; }
        public string City { get; set; }

        public string MissionDescription { get; set; }
        public string AvatarUrl { get; set; }

        public string SocialLinks { get; set; } // JSON string
        public bool IsPublic { get; set; }

        public int RoleId { get; set; }

        public DateTime? AgreedToTermsAt { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
    }
}

File: ClientApp/src/types/user-profile.ts
export interface UserProfile {
    id: string;
    userId: string;
    name: string;
    email: string;
    phone: string;
    country: string;
    city: string;
    missionDescription: string;
    avatarUrl: string | null;
    socialLinks: string; // JSON string (parse later)
    isPublic: boolean;
    roleId: number;
    agreedToTermsAt: string | null;
    createdAt: string;
    updatedAt: string;
}
